WordPress/gutenberg
# Polish experimental navigation block

## What?
This pull request polish a bit the code of [the experimental option to use the Interactivity API in the navigation block](https://github.com/WordPress/gutenberg/pull/50041).

## Why?

I used the pull request to clean the code a bit:
* Remove unnecessary directives.
* Remove unnecessary `async` calls.
* Use `requestUtils.createPage` instead of `admin.createPost` in the navigation tests.

## Testing Instructions
* Add the Page List block, and many submenus, to a Navigation block.
* Set the Overlay setting to be always off.
* Go to Gutenberg -> Experiments -> Activate Navigation block experimental option.
* Ensure the submenus and the Page List blocks are working as expected.
